retry/retry.go

```diff
@@ -11,27 +11,31 @@ import (
 	"github.com/pkg/errors"
 )
 
-const taskSystemName = "task"
+const (
+	taskSystemName = "task"
+	maxWaitTime    = 15 * float64(time.Minute) // 15 minutes
+)
 
 // Retry handles executing and retrying a function
 type Retry struct {
-	maxRetry uint // doesn't count initial try
+	maxRetry int // doesn't count initial try, set to -1 for infinite retries
```

Comment:
might be clearer to achieve this with a `*int` type, so it will just be set to `nil` to indicate no max retries

Reply:
why is it singular `retry` as opposed to `retries`?

Reply:
To me nil is the absence of something, so I equate nil with 0 which would be `no retries`

This was my reasoning for using -1 for infinite retries. I've also seen -1 used this way before

Reply:
Also I not sure why it was maxRetry, I left it from its original creation

Reply:
I named it "retry" - singular "retry" fits my intuition of how I think of this. If "retries" is better for other people, I'm ok to go with that.

I personally prefer -1 to avoid the additional check for nil when maxRetry is 0 or positive. It also would avoid any accidental nil pointer runtime error. Open to reasons for having a pointer though

Reply:
Sounds good, thank you both for reflecting on this.